neon_hana/auth/client_manager.py

```diff
@@ -23,74 +23,120 @@
 # LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
 # NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-from threading import Lock
+from uuid import uuid4
 
 import jwt
 
+from datetime import datetime
+from threading import Lock
 from time import time
 from typing import Dict, Optional
 from fastapi import Request, HTTPException
 from fastapi.security import HTTPBearer, HTTPAuthorizationCredentials
-from jwt import DecodeError
+from jwt import DecodeError, ExpiredSignatureError
 from ovos_utils import LOG
+from pydantic import ValidationError
 from token_throttler import TokenThrottler, TokenBucket
 from token_throttler.storage import RuntimeStorage
 
-from neon_hana.auth.permissions import ClientPermissions
+from neon_data_models.models.api.jwt import HanaToken
+from neon_hana.mq_service_api import MQServiceManager
+from neon_data_models.models.user import (User, NeonUserConfig,
+                                          PermissionsConfig)
+from neon_data_models.enum import AccessRoles
+from neon_hana.schema.auth_requests import AuthenticationResponse
+
+_DEFAULT_USER_PERMISSIONS = PermissionsConfig(klat=AccessRoles.USER,
+                                              core=AccessRoles.USER,
+                                              diana=AccessRoles.USER,
+                                              node=AccessRoles.USER,
+                                              hub=AccessRoles.USER,
+                                              llm=AccessRoles.USER)
 
 
 class ClientManager:
-    def __init__(self, config: dict):
+    def __init__(self, config: dict,
+                 mq_connector: Optional[MQServiceManager] = None):
         self.rate_limiter = TokenThrottler(cost=1, storage=RuntimeStorage())
 
-        self.authorized_clients: Dict[str, dict] = dict()
+        # Keep a dict of `client_id` to auth tokens that have authenticated to
+        # this instance
+        self._authorized_clients: Dict[str, AuthenticationResponse] = dict()
```

Comment:
It isn't persisted. Initially, I was considering caching this as a way to persist tokens, but with the user database now I don't see any practical application for this other than maybe approximating the number of clients using the backend.

It looks like it isn't used anymore and I [noted it may be worth deprecating](https://github.com/NeonGeckoCom/neon-hana/pull/33/files#diff-5073592301c4f8131e26ccf611121889ddd59295916473932a31e6f1fe2cfc62R90), so I will add a deprecation log to the property